Implemented: Added support for adding products while creating a TO using the barcode scanner (#1057)

R-Sourabh opened the pull request:
### Related Issues
<!--  Put related issue number which this PR is closing. For example #123 -->

#1057 

### Short Description and Why It's Useful
<!-- Describe in a few words what is this Pull Request changing and why it's useful -->
- Added support for adding products while creating the TO using the barcode scanner.
- Added the barcode scanning feature on the input.
- If a product is not matched, a `Match product` button is shown, which opens a modal on click to allow manual product selection.
- Added a scan button to start scanning mode; otherwise, the user can add items using input only, with proper handling for both cases.
- Enabled support for scanning items based on barcode scanning and allowed input search based on the barcodeIdentifier.
- Added the MatchProductModal to search for a product when no match is found.
- Added support to increment the quantity when scanning the same product using the barcode scanner, and added support to scroll to that product

### Screenshots of Visual Changes before/after (If There Are Any)
<!-- If you made any changes in the UI layer, please provide before/after screenshots -->
![image](https://github.com/user-attachments/assets/e5746c40-5931-4218-96f6-8a2a2ec2aad6)
![image](https://github.com/user-attachments/assets/656df242-9dee-45fd-b0f7-1f94692d11a8)
![image](https://github.com/user-attachments/assets/a7ff845c-3aa8-4b96-9e7e-665568ca8372)


### Contribution and Currently Important Rules Acceptance
<!-- Please get familiar with following info -->

- [x] I read and followed [contribution rules](https://github.com/hotwax/fulfillment#contribution-guideline)

R-Sourabh committed:
Implemented: Added support for adding products to the order while creating a TO using the barcode scanner (#1057)

R-Sourabh committed:
Improved: removed the unused imported icon (#1057)

R-Sourabh committed:
Improved: Added scan button to enable scanning and handled the case for adding a product using input separately (#1057)

R-Sourabh committed:
Improved: Added the translate function, updated the locale files, and renamed functions (#1057)

R-Sourabh committed:
Reverted: spacing change from the findProduct funtion(#1057)

R-Sourabh committed:
Improved: Updated function names, added a check to prevent the watcher from running, and populated searchedProduct when scanning is enabled (#1057)

R-Sourabh committed:
Improved: Added support for barcodeIdentifier when adding a product using scanning or input (#1057)

ymaheshwari1 commented on src/views/CreateTransferOrder.vue:
```diff
@@ -352,23 +373,103 @@ async function findProductFromIdentifier(payload: any) {
+
+// Updates the scanned product by checking if it already exists in the order and adding it if not
+async function scanProduct(scannedItem: string) {
+  if(!isScanningEnabled.value) return;
```
I think this check is not required as we only call this method when scanning is enabled, so we don't need to check the same condition again.

ymaheshwari1 commented on src/views/CreateTransferOrder.vue:
```diff
@@ -352,23 +373,103 @@ async function findProductFromIdentifier(payload: any) {
+async function scanProduct(scannedItem: string) {
+  if(!isScanningEnabled.value) return;
+  // Check if the product already exists in the order
+  const existingItem = currentOrder.value.items.find((item: any) => item.sku === scannedItem);
```
We should check the product against the barcode identifier here as well.

R-Sourabh committed:
Improved: Added a check in scanProduct to compare the scanned value with the barcode identifier (#1057)

- Changed the filters field value while scanning the product.
- Updated the locale file.

R-Sourabh committed:
Improved: Added trim() while searching for the item (#1057)

R-Sourabh committed:
Improved: Added the slot="end" attribute to the scan button (#1057)

R-Sourabh committed:
Improved: Updated the logic for removing an item when no matching product is found (#1057)

R-Sourabh committed:
Implemented: Added support to increment the quantity when scanning the same product using the barcode scanner, and added support to scroll to that product (#1057)

R-Sourabh committed:
Merge branch 'main' into #1057-barcode-scanner-for-transfer-orders
